### ISSUE TITLE
How to get features for all entities in online serving?

### ISSUE THREAD

**Issue by michcio1234** [feast-dev/feast#1361]
**Is your feature request related to a problem? Please describe.**
At inference time I would like to retrieve a dictionary of {entity_id: feature_value} for one given feature but for all entities stored currently in a feature table, **without knowing what IDs the entities have**.
E.g. I would like to retrieve `trips_today` for **all** the drivers that are currently in `driver_trips` table.

**Describe the solution you'd like**
Ideally, I could do:
```
features = client.get_online_features(
    feature_refs=["driver_trips:trips_today"],
    entity_rows=None,
).to_dict()
```
and get
```
{
    "driver_id": [... list of all the driver ids ...]
    "driver_trips:trips_today": [... list of their respective feature values ...]
}
```

**Describe alternatives you've considered**
🤷🏻‍♂️ 

**Additional context**
I realise it's kind of using Feast as a database. If this is totally out of Feast philosophy, please enlighten me. In such case, what alternative could I use?

Thanks.

**Comment by woop** [comment]
@michcio1234 Your requirement is one I have heard before, it's definitely not uncommon. It's one that is very hard for us to support with databases like Dynamo/Bigtable/Redis since it requires scanning over keys or maintaining an index ourselves. We aren't yet sure whether we want to support this functionality ever, although it's one that we are super actively discussing right now.

**Comment by lucasbmiguel** [comment]
This same issue occurs for the get_historical_feature right?

**Comment by woop** [comment]
> This same issue occurs for the get_historical_feature right?

Yes, the same limitation is on the `get_historical_features()` method, although in that case we are more open to having range scans over an offline store.

For what it's worth @michcio1234, there is an active discussion on the Feast mailing list around this functionality. It would be great if you could join that discussion and throw in your 2c.
